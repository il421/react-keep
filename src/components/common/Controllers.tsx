import React from "react";
import { AlignItems, JustifyContent } from "../../common";
import { FlexBox, IconButton } from "../ui-components";
import "../../styles/components/common/_controllers.scss";
import { NoteType } from "../notes";

interface Controllers extends Partial<Pick<HTMLElement, "className">> {
  isMobile?: boolean;
  openDialog: (type: NoteType) => void;
}

export const Controllers: React.FunctionComponent<Controllers> = ({
  className,
  isMobile = false,
  openDialog,
}) => {
  const handleOnClick = (type: NoteType) => () => openDialog(type);

  return (
    <div
      className={`controllers  ${
        isMobile && "controllers--mobile"
      } ${className}`}
    >
      <FlexBox
        vertical={isMobile}
        justifyContent={JustifyContent.spaceBetween}
        alignItems={AlignItems.center}
      >
        <IconButton
          className="controllers__text"
          onButtonClick={handleOnClick(NoteType.text)}
          icon="align-left"
          size="2x"
        />

        <IconButton
          className="controllers__list"
          onButtonClick={handleOnClick(NoteType.list)}
          icon="check-square"
          size="2x"
        />

        <IconButton
          className="controllers__image"
          onButtonClick={handleOnClick(NoteType.image)}
          icon="image"
          size="2x"
        />
      </FlexBox>
    </div>
  );
};
