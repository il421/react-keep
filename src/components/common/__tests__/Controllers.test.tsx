import { shallow, ShallowWrapper } from "enzyme";
import React from "react";

import { Controllers } from "../Controllers";

test("should render Controllers correctly", () => {
  const wrapper: ShallowWrapper<typeof Controllers> = shallow<
    typeof Controllers
  >(<Controllers openDialog={jest.fn} />);
  expect(wrapper.debug()).toMatchSnapshot();
});
