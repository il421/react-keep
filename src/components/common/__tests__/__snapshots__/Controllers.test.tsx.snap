// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Controllers correctly 1`] = `
"<div className=\\"controllers  false undefined\\">
  <FlexBox vertical={false} justifyContent=\\"space-between\\" alignItems=\\"center\\">
    <IconButton className=\\"controllers__text\\" onButtonClick={[Function: onButtonClick]} icon=\\"align-left\\" size=\\"2x\\" />
    <IconButton className=\\"controllers__list\\" onButtonClick={[Function: onButtonClick]} icon=\\"check-square\\" size=\\"2x\\" />
    <IconButton className=\\"controllers__image\\" onButtonClick={[Function: onButtonClick]} icon=\\"image\\" size=\\"2x\\" />
  </FlexBox>
</div>"
`;
